### ISSUE TITLE
Add an optional parameter sort_name to the join request

### ISSUE THREAD

**Issue by ffdixon** [bigbluebutton/bigbluebutton#16813]
**Is your feature request related to a problem? Please describe.**
When exporting the user list to a text file, we are currently sorting by first letter and by last name.  However,  when a front-end passes a name such as

```
  Fred Dixon - Product Manager
```

the "last name" becomes "Manager", which isn't correct.

**Describe the solution you'd like**
Add a new parameter to the `join` command called `sort_name`, which gives BigBlueButton the name for sorting.  For example, 

```
&sort_name=Dixon,%20Fred
```

which allows BigBlueButton to correctly sort the names based on how the front-end wishes.  (In this case, by last name then first name"

**Describe alternatives you've considered**
We could try to introduce a special separator character, such as `|` for separating additional information from the name, so if the front end passes the following as `fullName` 

```
  Fred Dixon | Product Manager
```

BigBlueButton would split at the `|` character and the last name would become `Dixon`.

However, it's arbitrary and not very reliable, and wouldn't localize very well.  It's much more deterministic to give the front-end the ability to specify the exact sort order by passing a string in `sort_name`

**Comment by gustavotrott** [comment]
Currently, we have the `nameSortable` field, which represents the user's name in lowercase and stripped of any accents.

To introduce additional details without causing confusion, I recommend adding a new parameter such as `name-detail`, `name-suffix`, `name-extension`, or `name-extra-info`. This parameter would allow us to include supplementary information associated with the user's name.

For example, if we have `name=Fred Dixon` and `name-detail=Product Manager`, BBB would display this in the user list as "Fred Dixon, Product Manager". This approach ensures that the primary `name` field remains uncluttered and retains its ability to extract the last name efficiently.

**Comment by antobinary** [comment]
> For example, if we have name=Fred Dixon and name-detail=Product Manager, BBB would display this in the user list as "Fred Dixon, Product Manager".

In this case we'd need to guess which divider makes sense to use, which order etc.
I think it's clearer to use the string passed by the front end in its exact format (in the userlist) and if there is desire for the sorting to do a proper job - then a new field. To avoid confusion with the existing internal `nameSortable` we could rename the new parameter, for example `cleanName` which in frontends could be used as
`cleanName: person's name without titles, position etc. Used for sorting users purpose`
Here "clean" is inspired by functionalities like "Copy clean link" for example where various tailing args are removed.
Alternatives are welcome, the above is a personal opinion.

Another option:
if we sort by first vs last name, perhaps we could introduce `firstName` (optional), `lastName` (optional), in addition to `fullName` (required). If either/both first and last name are provided, use those for sorting purposes. That would eliminate the need for us to guess or split at the right point. (The original proposition of `sort_name` would have us guessing where to split if we had more than one space/divider in the string.)

**Comment by gustavotrott** [comment]
> In this case we'd need to guess which divider makes sense to use, which order etc. 

Instead of relying on a fixed separator or order, we can make the display format flexible by introducing a `userNameTemplate` config. For example:

```
userNameTemplate: "%FULLNAME% - %NAME_DETAIL%"
```

By keeping `nameDetail` in a separate field, we achieve several benefits:

- The design team can utilize `nameDetail` in various ways across different parts of the system. 
Not the ideal, but one quick example:

![image](https://github.com/user-attachments/assets/f6e42438-972f-489a-ab02-3b6035656748)

- The user list might display "Fred Dixon, Product Manager," while other sections like "Who is Talking" could show only  "Fred Dixon".
- `fullName` remains uncluttered, preserving its integrity for tasks like searching.
- Administrators can choose the appropriate separator in the `userNameTemplate`, avoiding the need to guess or parse the separator in the backend.


I know your idea also tackle the main issue, but I see the field `nameDetail` as more intuitive what it means than a `cleanName` field. A quick comparison:

**Your Initial Idea:**
  - `fullName: Fred Dixon - Product Manager`
  - `cleanName: Fred Dixon`
  
  Pros:
  - We don't need to guess the separator or order

**My Idea:**
  - `fullName: Fred Dixon`
  - `nameDetail: Product Manager`
  - `userNameTemplate: "%FULLNAME% - %NAME_DETAIL%"`
  
  Pros:
  - Keeps additional information separate for versatile usage.
  - Allows the design team to format the display as needed without backend constraints.

**Comment by gustavotrott** [comment]
> Another option: if we sort by first vs last name, perhaps we could introduce `firstName` (optional), `lastName` (optional), in addition to `fullName` (required). If either/both first and last name are provided, use those for sorting purposes. That would eliminate the need for us to guess or split at the right point. (The original proposition of `sort_name` would have us guessing where to split if we had more than one space/divider in the string.)

I like the idea of being able to specify the `firstName` and `lastName` fields. Even with a "cleanName" field, extracting the last name is always a guess. Some people have two last names, and the system cannot deduce that, often using only the last word as the surname.  

I prefer this approach over the original one suggested in the issue.  

That said, I would still advocate for a `nameDetail` field for added flexibility and greater clarity (avoiding mixing names and professions in a single field). 
However, we can start with this idea by adding just these two fields if that's your preference!  

Initially, these new fields would be used exclusively for the purpose of exporting users, correct?

**Comment by antobinary** [comment]
>  we can start with this idea by adding just these two fields if that's your preference!

Yes please :pray: !  Phase one -- add `firstName` (optional) and `lastName` (optional) + updating the docs.
Phase two - adding `nameDetail` (optional) and `userNameTemplate` config.
